simulation/traffic_simulator/src/entity/entity_manager.cpp

```diff
@@ -551,34 +551,48 @@ auto EntityManager::toMapPose(const openscenario_msgs::msg::LaneletPose & lanele
   return hdmap_utils_ptr_->toMapPose(lanelet_pose).pose;
 }
 
+openscenario_msgs::msg::EntityStatus EntityManager::updateNpcLogic(
+  const std::string & name,
+  const std::unordered_map<std::string, openscenario_msgs::msg::EntityType> & type_list)
+{
+  if (verbose_) {
+    std::cout << "update " << name << " behavior" << std::endl;
+  }
+  entities_[name]->setEntityTypeList(type_list);
+  entities_[name]->onUpdate(current_time_, step_time_);
+  if (entities_[name]->statusSet()) {
+    return entities_[name]->getStatus();
+  }
+  THROW_SIMULATION_ERROR("status of entity ", name, "is empty");
+}
+
 void EntityManager::update(const double current_time, const double step_time)
 {
   std::chrono::system_clock::time_point start, end;
   start = std::chrono::system_clock::now();
   step_time_ = step_time;
   current_time_ = current_time;
-  if (current_time_ >= 0) {
-    traffic_light_manager_ptr_->update(step_time_);
-  }
   if (verbose_) {
     std::cout << "-------------------------- UPDATE --------------------------" << std::endl;
     std::cout << "current_time : " << current_time_ << std::endl;
   }
   if (getNumberOfEgo() >= 2) {
     THROW_SEMANTIC_ERROR("multi ego simulation does not support yet");
   }
+  if (current_time_ >= 0) {
+    traffic_light_manager_ptr_->update(step_time_);
+  }
   setVerbose(verbose_);
   auto type_list = getEntityTypeList();
   std::unordered_map<std::string, openscenario_msgs::msg::EntityStatus> all_status;
-  for (auto it = entities_.begin(); it != entities_.end(); it++) {
-    if (verbose_) {
-      std::cout << "update " << it->first << " behavior" << std::endl;
-    }
-    it->second->setEntityTypeList(type_list);
-    it->second->onUpdate(current_time, step_time);
-    if (it->second->statusSet()) {
-      auto status = it->second->getStatus();
-      all_status.emplace(it->first, status);
+  const std::vector<std::string> entity_names = getEntityNames();
+  std::unordered_map<
+    std::string, std::shared_ptr<std::future<openscenario_msgs::msg::EntityStatus>>>
+    futures;
```

Comment:
The variable future is not used, so delete it.

Reply:
I forget to delete this line, thanks.